### PR TITLE
Split configuration and add resetPeerList for region migration

### PR DESCRIPTION
**Pull request by BUAAserein**
## Description

Split configuration and add resetPeerList for region migration, see [feishu-doc](https://apache-iotdb.feishu.cn/docx/MC8gdgTgOoG9eYx9MIpcrtB9n4e) for details.

### CODE REVIEW

File: iotdb-core/consensus/src/main/java/org/apache/iotdb/consensus/iot/IoTConsensusServerImpl.java [hunk 1]
```diff
@@ -635,20 +623,59 @@ public void recoverConfiguration() {
       // interrupted
       // unexpectedly, we need substitute configuration with tmpConfiguration file
       if (Files.exists(tmpConfigurationPath)) {
-        if (Files.exists(configurationPath)) {
-          Files.delete(configurationPath);
-        }
+        Files.deleteIfExists(configurationPath);
         Files.move(tmpConfigurationPath, configurationPath);
       }
       buffer = ByteBuffer.wrap(Files.readAllBytes(configurationPath));
```
Comment (liyuheng55555) [inline iotdb-core/consensus/src/main/java/org/apache/iotdb/consensus/iot/IoTConsensusServerImpl.java:629]: After the configuration.dat file is split, seems Files.readAllBytes(configurationPath) will cause `IOException`, then line 637 will catch this exception and handle it with the new `getConfiguration` method.
Thus, there are some other code also throw `IOException`. This makes the code here somewhat unclear.
image
It is hoped to use a clearer method to determine whether the current processing is a new version file or an old version file (for example, whether the CONFIGURATION_FILE_NAME file exists or not).

**Review by liyuheng55555 — CHANGES_REQUESTED** [review]
PTAL~

File: iotdb-core/consensus/src/main/java/org/apache/iotdb/consensus/iot/IoTConsensusServerImpl.java (hunk 1 above)
Comment (BUAAserein) [reply iotdb-core/consensus/src/main/java/org/apache/iotdb/consensus/iot/IoTConsensusServerImpl.java]: ok.

**Review by liyuheng55555 — APPROVED** [review]
LGTM

**Pull request merged**